### PR TITLE
feat: extend PickerNetwork component functionality

### PR DESCRIPTION
**Pull request by vinnyhoward**
## **Description**

This PR changes the `PickerNetwork` to allow the network name to be hidden. These changes are necessary to accommodate the [Header Update](https://www.figma.com/design/aMYisczaJyEsYl1TYdcPUL/Portfolio-View?m=auto&node-id=5019-59596&t=PAdxL1bg2Mk08dSk-1) design requirements coming up. However the aim is to introduce very little change (other than padding) to what is pre-existing but by passing in the prop of `hideNetwork` it changes into the future design

## **Related issues**

Related: [#11763](https://github.com/MetaMask/metamask-mobile/pull/11763)

## **Manual testing steps**

1. View any instance of `PickerNetwork` being used. For example the `PickerNetwork` in header in the home screen

## **Screenshots/Recordings**

before

After `hideNetwork=false` the default
after (no prop)

After `hideNetwork=true`
after (prop)

### **Before**

NA

### **After**

NA

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/component-library/components/Pickers/PickerNetwork/PickerNetwork.styles.ts [hunk 1]
```diff
@@ -28,7 +28,7 @@ const styleSheet = (params: {
       {
         height: 32,
         borderRadius: 16,
-        paddingHorizontal: 8,
+        paddingHorizontal: 12,
```
Comment (brianacnguyen) [inline app/component-library/components/Pickers/PickerNetwork/PickerNetwork.styles.ts:31]: why was this padding changed?

File: app/component-library/components/Pickers/PickerNetwork/PickerNetwork.styles.ts [hunk 2]
```diff
@@ -40,6 +40,9 @@ const styleSheet = (params: {
       marginHorizontal: 8,
       flexShrink: 1,
     },
+    networkIconContainer: {
+      marginRight: 8,
+    },
```
Comment (brianacnguyen) [inline app/component-library/components/Pickers/PickerNetwork/PickerNetwork.styles.ts:45]: can you show this change?

**Review by brianacnguyen — COMMENTED** [review]
1 small comment

File: app/component-library/components/Pickers/PickerNetwork/PickerNetwork.styles.ts (hunk 2 above)
Comment (vinnyhoward) [reply app/component-library/components/Pickers/PickerNetwork/PickerNetwork.styles.ts:45]: So when passing the prop `hideNetwork` as `true` the style `networkIconContainer` will be passed in to adjust the icon distance from the network avatar.

Without the new style
Screenshot 2024-10-22 at 12 55 42 PM

With the new style
Screenshot 2024-10-22 at 12 56 26 PM

File: app/component-library/components/Pickers/PickerNetwork/PickerNetwork.styles.ts (hunk 1 above)
Comment (vinnyhoward) [reply app/component-library/components/Pickers/PickerNetwork/PickerNetwork.styles.ts]: After reviewing the UI changes I decided to revert this. After another look I think it matches the Figma slightly more

**Review by gambinish — APPROVED** [review]

**Review by georgeweiler — APPROVED** [review]

**Review by darkwing — APPROVED** [review]

**Review by brianacnguyen — APPROVED** [review]

**Pull request merged**